src/unity/python/turicreate/toolkits/activity_classifier/_activity_classifier.py

```diff
@@ -165,7 +166,12 @@ def create(dataset, session_id, target, features=None, prediction_window=100,
     _tkutl._raise_error_if_sarray_not_expected_dtype(dataset[session_id], session_id, [str, int])
 
     if isinstance(validation_set, str) and validation_set == 'auto':
-        dataset, validation_set = _random_split_by_session(dataset, session_id)
+        unique_sessions = _SFrame({'session': dataset[session_id].unique()})
```

Comment:
This calculation of the number of sessions is actually quite heavy, and takes a relatively very long time for larger datasets. That's the main reason we've added a `num_sessions` output to `_prep_data()`.

As you've mentioned in https://github.com/apple/turicreate/issues/964#issuecomment-413922347 it would probably be a better idea to do the train-test split as part of the chunking step.

Reply:
Good idea, I added a TODO linking to #991